pyramid_openapi3/exceptions.py

```diff
@@ -35,5 +35,74 @@ def __str__(self) -> str:
         """Return str(self.detail) or self.explanation."""
         return str(self.detail) if self.detail else self.explanation
 
-    def _json_formatter(self, status, body, title, environ) -> t.Dict:
-        return {"message": body, "code": status, "title": self.title}
+
+def extract_error(request: Request, err: OpenAPIError) -> t.Dict[str, str]:
+    """Extract error's JSON response.
+
+    You can tell pyramid_openapi3 to use your own version of this
+    function by providing a dotted-name to your function in
+    `request.registry.settings["pyramid_openapi3_extract_error"]`.
+
+    This function expects the below definition in openapi.yaml
+    file. If your openapi.yaml is different, you have to
+    provide your own extract_error() function.
+
+    ```
+    components:
+
+      schemas:
+
+        Error:
+          type: object
+          required:
+            - exception
+            - message
+          properties:
+            field:
+              type: string
+            message:
+              type: string
+            exception:
+              type: string
+
+      responses:
+
+        ValidationError:
+          description: OpenAPI request/response validation failed
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Error"
+    ```
+    """
+    if getattr(err, "schema_errors", None) is not None:
+        for schema_error in err.schema_errors:  # pragma: no branch
+            return extract_error(request, schema_error)
```

Comment:
I didn't check what err.schema_errors looks like but the code reads like that (and maybe I get a bit fuzzed about it, so please ignore if you disagree)

The loop iterates over `err.schema_errors` and returns the first one in this list. In case there are more items in `err.schema_errors they`` get ignored.

No sure whether this is intended behaviour, but if yes, maybe writing it like this would make the intent obvious?
```
if getattr(err, "schema_errors", None):
  return extract_error(request, err.schema_error[0])
```
should be exactly the same and makes the intention clear?

Reply:
Let me look into this one more time.